tests/macsec/test_rekeying.py

```diff
@@ -0,0 +1,87 @@
+import pytest
+import logging
+import json
+import sys
+import time
+import os
+
+if sys.version_info.major > 2:
+    from pathlib import Path
+    sys.path.insert(0, str(Path(__file__).parent))
+
+from .macsec_config_helper import setup_macsec_configuration
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.macsec_required,
+    pytest.mark.topology("any")
+]
+
+profile_name = "256_XPN_SCI"
+new_rekey_period = 180
+
+
+@pytest.fixture(scope='module')
+def setup(duthost, macsec_nbrhosts, ctrl_links):
```

Comment:
@parmarkj can you add T2 support for this testcase?
here in the setup you could use `duthosts` and `enum_rand_one_per_hwsku_frontend_hostname`